### PR TITLE
Removed almost all deprecations on Product page

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | The logger is full of deprecations messages (almost 250 on product page).
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | In dev mode, the number of deprecations should be under 5 (mainly only the deprecated calls to the logger should remain, as it's already fixed in another [pull request](https://github.com/PrestaShop/PrestaShop/pull/8710)) 

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8864)

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Do we have to remove all occurencies of `choice_as_value`?
In https://github.com/PrestaShop/PrestaShop/pull/8705/files#diff-3891386db15ebba1fb39d64a63e6a127R48, I'm adding it to the new SwitchType

**Review by Quetzacoalt91 — DISMISSED** [review]

**Comment by mickaelandrieu** [comment]
> Do we have to remove all occurrences of choice_as_value?

Yes, by default the value is true and this property doesn't exist in Symfony 4 (boring, cause we have to add it to make the forms works in Symfony 3 according to the old Symfony 2.8 deprecation system :cry: .

File: src/PrestaShopBundle/Resources/config/admin/services_alias.yml [hunk 1]
```diff
@@ -2,12 +2,9 @@ services:
     # Service aliases for backward campatibility
     prestashop.core.api.stockMovement.repository:
         alias: prestashop.core.api.stock_movement.repository
-        deprecated: 'The "%service_id%" service is deprecated. Use "prestashop.core.api.stock_movement.repository" instead'
```
Comment (Quetzacoalt91) [inline src/PrestaShopBundle/Resources/config/admin/services_alias.yml]: Is this `deprecated` label now deprecated? Or you just don't like it? :)

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Resources/config/admin/services_alias.yml]: it's deprecated since Symfony 2.8 xD

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**